### PR TITLE
btcjson: Add getmempoolentry API

### LINKED ISSUE #908: btcjson: Add getmempoolentry API

### PR DESCRIPTION
**Pull request by dajohi**
Fixes #908

### CODE REVIEW

**Comment by FrozenPrincess** [comment]
<3

BTW the struct `GetRawMempoolVerboseResult` and `GetMempoolEntryResult` should actually be the same thing (although at the moment GetRawMempoolVerboseResult is missing fields), so it might be best to typedef them or something

**Comment by dajohi** [comment]
Thanks for pointing that out.  I'll do that after btcd gets support for the new fields in getrawmempool verbose.

**Comment by dajohi** [comment]
Fixes #908

File: btcjson/chainsvrcmds.go [hunk 1]
```diff
@@ -333,6 +333,19 @@ func NewGetInfoCmd() *GetInfoCmd {
 	return &GetInfoCmd{}
 }
 
+// GetMempoolEntryCmd defines the getmempoolentry JSON-RPC command.
+type GetMempoolEntryCmd struct {
+	TxID string
```
Comment (jrick) [inline btcjson/chainsvrcmds.go]: doesn't need it actually. json-rpc requests are positional, not keyed in an object.

Comment (davecgh) [inline btcjson/chainsvrcmds.go]: Ah right, I noticed it was on one of the others (`GetHeadersCmd`).  Given it's positional though, you're right it isn't needed.

Comment (jrick) [inline btcjson/chainsvrcmds.go]: Yep.. I added it to that on accident.  We can remove it but it doesn't really matter, it just gets ignored.

**Pull request merged**